pkg/dockerregistry/server/metrics/metrics.go

```diff
@@ -60,24 +94,63 @@ func (m *metricTimer) Stop() {
 	m.collector.WithLabelValues(m.labels...).Observe(float64(time.Since(m.startTime)) / float64(time.Second))
 }
 
-func newWrapper(reponame string) wrapped.Wrapper {
-	return func(ctx context.Context, funcname string, f func(ctx context.Context) error) error {
-		defer NewTimer(registryAPIRequests, []string{strings.ToLower(funcname), reponame}).Stop()
+// NewRepository wraps a distribution.Repository to collect statistics.
+func NewRepository(r distribution.Repository, reponame string) distribution.Repository {
+	return wrapped.NewRepository(r, func(ctx context.Context, funcname string, f func(ctx context.Context) error) error {
+		defer NewTimer(requestDurationSeconds, []string{strings.ToLower(funcname), reponame}).Stop()
 		return f(ctx)
+	})
+}
+
+func dockerErrorCode(err error) string {
+	if e, ok := err.(errcode.Error); ok {
+		return e.ErrorCode().String()
 	}
+	return "UNKNOWN"
 }
 
-// NewBlobStore wraps a distribution.BlobStore to collect statistics.
-func NewBlobStore(bs distribution.BlobStore, reponame string) distribution.BlobStore {
-	return wrapped.NewBlobStore(bs, newWrapper(reponame))
+// NewPullthroughRepository wraps distribution.Repository that is instantiated
+// for a remote registry to collect statistics.
+func NewPullthroughRepository(r distribution.Repository, registry string) distribution.Repository {
+	return wrapped.NewRepository(r, func(ctx context.Context, funcname string, f func(ctx context.Context) error) error {
+		labelValues := []string{strings.ToLower(registry), strings.ToLower(funcname)}
```

Comment:
based on discussion here: https://github.com/openshift/origin/pull/19248#issuecomment-388962019

let's not lowercase these values, just use them as is.